### PR TITLE
[Improvement](predicate) optimize short-circuit predicates

### PR DESCRIPTION
**Pull request by Gabriel39**
# Proposed changes

For scan node with no vectorized predicate, the input column for the first short-circuit predicate is dense and we don't need to access the selector column.

This PR improve performance by ~30%  on TPCH Q3.

## Problem summary

Describe your changes.

## Checklist(Required)

* [ ] Does it affect the original behavior
* [ ] Has unit tests been added
* [ ] Has document been added or modified
* [ ] Does it need to update dependencies
* [ ] Is this PR support rollback (If NO, please explain WHY)

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by Gabriel39** [comment]
run buildall

**Comment by Gabriel39** [comment]
run feut

**Comment by Gabriel39** [comment]
run beut

File: be/src/olap/null_predicate.cpp [hunk 1]
```diff
@@ -54,10 +54,17 @@ uint16_t NullPredicate::evaluate(const vectorized::IColumn& column, uint16_t* se
...
+            for (uint16_t i = 0; i < size; ++i) {
+                sel[new_size] = i;
+                new_size += (null_map[i] == _is_null);
+            }
+        } else {
+            for (uint16_t i = 0; i < size; ++i) {
+                uint16_t idx = sel[i];
+                sel[new_size] = idx;
+                new_size += (null_map[idx] == _is_null);
+            }
```
Comment (HappenLee) [inline be/src/olap/null_predicate.cpp:67]: better need a function to do:
```
uint16_t idx = is_parse_column ? i : sel[i];
```
to simple the function logic

`is_parse_column` better be const bool

**Comment by Gabriel39** [comment]
run buildall

**Review by yiguolei — APPROVED** [review]
LGTM

**Pull request merged**